### DISCUSSION TITLE
Importing Dashboard JSON should prompt to fix missing data sources

### DISCUSSION THREAD

**Discussion by joshhunt** [grafana/grafana#38320] [Dashboards, Panels & Visualizations]
When I import dashboard JSON, the configured data sources may be named differently than where it was exported from, so the panels would not load any data until I go into each panel and adjust the datasource.

**What would you like to be added**:

If the data sources in a JSON import are not in the grafana instance, the import UI should prompt me to fix up any missing dashboards, or leave them as is.

Preferably it could do a "replace all" to fix all missing data sources of the same name in one swoop, then perhaps with individual overrides? (or just leave that to going back and editing each panel)

**Why is this needed**:

To make importing dashboards easier :)

**Comment by marefr** [comment]
Did you export the dashboards for sharing externally before importing them?

**Comment by joshhunt** [comment]
@marefr I'm not sure how they were exported to be honest - I imported the dashboard "manually" from the provisioning repo.

**Comment by marefr** [comment]
If the dashboard is not exported for sharing externally Grafana cannot prompt for missing data sources since Grafana doesn't no what data source type is expected. 

Not sure if anything can be made to improve that short-term. I'll ask @grafana/grafana-frontend-platform for input

**Comment by joshhunt** [comment]
I think it still can, no? I think Grafana should know at least that the data source specified in the dashboard is missing in the user's Grafana, and it should be able to prompt to "fix" the data sources.

Even if there's no type specified, i think prompting to fix from all available data sources is an improvement on "successfully" importing a dashboard, but having every panel immediately error out and needing to fix manually panel-by-panel.

**Comment by yurividal** [comment]
I second this. I have some "template" dashboards that i constantly import to different locations. And, every time I need to paste the json in a notepad, and manually find and replace the id of the datasources. This is a workaround that works when my datasource is infuxdb. When its elasticsearch, this workaround doesn't work, and all my panels that reference elasticsearch are reset to a default blank query. Its a pain.

**Comment by dwt** [comment]
I've also just encountered this - here we have different k8s clusters (dev/uat/prod) that we want our dashboards to flow through. And I'm not at all sure how to ameliorate this.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
